frameworks/Java/play2-java/play2-java-jooq-hikaricp/app/controllers/Application.java

```diff
@@ -0,0 +1,121 @@
+package controllers;
+
+import static models.tables.Fortune.FORTUNE;
+import static models.tables.World.WORLD;
+import static play.mvc.Http.MimeTypes.JSON;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.inject.Inject;
+
+import org.jooq.DSLContext;
+import org.jooq.JSONFormat;
+import org.jooq.JSONFormat.RecordFormat;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.jooq.types.UInteger;
+
+import models.tables.records.FortuneRecord;
+import models.tables.records.WorldRecord;
+import play.db.Database;
+import play.mvc.Controller;
+import play.mvc.Result;
+import utils.DatabaseExecutionContext;
+
+public class Application extends Controller {
+
+    private static final SQLDialect DIALECT = SQLDialect.MYSQL_5_7;
+    private static final JSONFormat JSON_FORMAT = JSONFormat.DEFAULT_FOR_RECORDS.recordFormat(RecordFormat.OBJECT);
+
+    private final Database db;
+    private final DatabaseExecutionContext dbEc;
+
+    @Inject
+    public Application(final Database db, final DatabaseExecutionContext dbEc) {
+        this.db = db;
+        this.dbEc = dbEc;
+    }
+
+    public CompletionStage<Result> db() {
+        return getRandomWorlds(1).thenApply(worlds -> ok(worlds.get(0).formatJSON(JSON_FORMAT)).as(JSON));
+    }
+
+    public CompletionStage<Result> queries(final String queries) {
+        return getRandomWorlds(queryCount(queries)).thenApply(worlds -> ok(worlds.formatJSON(JSON_FORMAT)).as(JSON));
+    }
+
+    public CompletionStage<Result> fortunes() {
+        return CompletableFuture.supplyAsync(() -> {
+            final List<FortuneRecord> fortunes = this.db.withConnection(connection -> {
+                return DSL.using(connection, DIALECT).selectFrom(FORTUNE).fetch();
+            });
+            fortunes.add(new FortuneRecord(UInteger.valueOf(0), "Additional fortune added at request time."));
+            Collections.sort(fortunes, (f1, f2) -> f1.getMessage().compareTo(f2.getMessage()));
+
+            return ok(views.html.fortunes.render(fortunes));
+        }, dbEc);
+    }
+
+    public CompletionStage<Result> update(final String queries) {
+        return getRandomWorlds(queryCount(queries)).thenApplyAsync(worlds -> {
+            final Random random = ThreadLocalRandom.current();
+            for (final WorldRecord world : worlds) {
+                world.setRandomnumber((random.nextInt(10000) + 1));
+            }
```

Comment:
Again, what is being benchmarked here? The database access layer, or some random number generator? All of this stuff can lead to bias in a benchmark result - not sure if that's relevant, of course...

Reply:
This is a test requirement.
Also, that is the same code like for the jpa and ebean tests, so it's fair.